R/msnset2se.R

```diff
@@ -30,19 +31,33 @@ se2msnset <- function(se) {
 #' @param msnset MSnSet object,
 #' Object which will be turned into an SummarizedExperiment object.
 #' @return SummarizedExperiment object.
-#' @export
 msnset2se <- function(msnset) {
   # Check input
-  assertthat::assert_that(inherits(msnset, "MSnSet"))
+  stopifnot(inherits(msnset, "MSnSet"))
 
+	if(!requireNamespace("SummarizedExperiment")) {
+		stop("The SummarizedExperiment package is required for MSnSet to SummarizedExperiment conversion.")
+	}
+	
   # Extract assay, rowData and colData
-  raw <- Biobase::exprs(msnset)
-  row_data <- Biobase::fData(msnset)
-  col_data <- Biobase::pData(msnset)
+  raw <- exprs(msnset)
+  rowData <- fData(msnset)
+  colData <- pData(msnset)
 
   # Generate SE
-  se <- SummarizedExperiment::SummarizedExperiment(assays = as.matrix(raw),
-                                                   rowData = row_data,
-                                                   colData = col_data)
-  return(se)
+  SummarizedExperiment::SummarizedExperiment(assays = as.matrix(raw),
+  																					 rowData = rowData,
+  																					 colData = colData)
 }
+
+setAs("MSnSet", 
+			"SummarizedExperiment",
+			function(from) {
+				msnset2se(from)
```

Comment:
The functions are very short and not used anywhere else. That's why I don't see the point of an additional internal function. Please put the function body into the `setAs` method.

Reply:
@arnesmits - I believe you use use `se2msnset` and `msnset2se` in your package and possibly mention them in documentation or vignette. I think we can keep them, at least in a first instance, as they might already be used here and there. But the official `as` syntax should probably he preferred.

Reply:
I am against exporting `se2msnset` and `msnset2se`:

1. We prefer the `as` syntax.
2. `x2y` is uncommon.
2. If we add it we would teach the user a "wrong" approach.
3. If a second package (`MSnbase`) supports them they become more common and seem more "correct".
4. To deprecate/defunct a function is effort (we have to remember it for more than one release cycle, documentation, ...) and we maybe won't ever do it.

Reply:
@arnesmits - are you ok with marking the old forms as Defunct and suggest to use the standard `as` form, so we can redirect users that might already have used them to a more standard form?

Reply:
I agree with both of you that the common _as_ syntax would be preferable. The old form is mentioned in the documentation of my package, although it is not required in the default workflow. I will mark it as Defunct and suggest the _as_ form, once this is incorporated in MSnbase.